### PR TITLE
Feature/ps 63 bo show charge for dispute

### PR DESCRIPTION
**Pull request by sadtuna**
https://shore.myjetbrains.com/youtrack/issue/PS-63

### CODE REVIEW

File: lib/payment/response_classes/dispute.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ def created_at=(val)
       @created_at = DateTime.parse(val).to_date
     end
 
+    def charge
+      MerchantConnector.new(@merchant_id).get_charge(@charge_id)
```
Comment (ceigel) [inline lib/payment/response_classes/dispute.rb]: Please use caching for this. 

```ruby
@charge ||= MerchantConnector.new(@merchant_id).get_charge(@charge_id)
```

Comment (sadtuna) [inline lib/payment/response_classes/dispute.rb]: sounds good idea

File: lib/payment/response_classes/merchant_stripe.rb [hunk 2]
```diff
@@ -188,7 +188,8 @@ def <=>(other)
     end
 
     attr_accessor :charge_id, :status, :amount_cents, :currency,
-                  :customer_name, :credit_card_brand, :created_at
+                  :customer_name, :customer_email, :credit_card_brand,
```
Comment (ceigel) [inline lib/payment/response_classes/merchant_stripe.rb]: This shouldn't be added here. The payment service returns the customer_email only as a detail. And we don't use this class in that case.

Comment (sadtuna) [inline lib/payment/response_classes/merchant_stripe.rb]: I think I have to add here all the detailed response attributes too, if we want to use this class to represent:

* GET charge response (not needed in core now)
* GET dispute response's charge part (we need only the customer_email now)

File: spec/payment/merchant_stripe_spec.rb [hunk 3]
```diff
@@ -386,7 +386,8 @@
         status: 'succeeded',
         amount_cents: 1000,
         currency: 'eur',
-        customer_name: 'Jane Austion',
+        customer_name: 'Jane Austen',
+        customer_email: 'jane.austen.1775@gmail.com',
```
Comment (ceigel) [inline spec/payment/merchant_stripe_spec.rb]: These should be removed...

**Comment by ceigel** [comment]
I thought you need the customer_email and other fields in the charge object...

File: lib/payment/response_classes/merchant_stripe.rb [hunk 4]
```diff
@@ -188,7 +198,9 @@ def <=>(other)
     end
 
     attr_accessor :charge_id, :status, :amount_cents, :currency,
-                  :customer_name, :credit_card_brand, :created_at
+                  :customer_name, :credit_card_brand, :created_at,
+                  :amount_refunded_cents, :customer_address, :customer_email,
+                  :credit_card_last4, :description, :services
```
Comment (ceigel) [inline lib/payment/response_classes/merchant_stripe.rb]: Maybe you could group them based on scope. Ex:
```ruby
            :customer_name, :customer_address, :customer_email,
            :credit_card_brand, :credit_card_last4
```

File: spec/support/stripe_payment.rb [hunk 5]
```diff
@@ -139,6 +139,9 @@ def payment_service_dispute_response(mid, attributes)
       'has_evidence' => true,
       'past_due' => false,
       'submission_count' => 0,
+      'charge' => {
+        'charge_id' => 'ch_17kyvuBJMmId6xqIDWIRAimq'
```
Comment (ceigel) [inline spec/support/stripe_payment.rb]: Please add an 'object' => 'charge'

**Comment by ceigel** [comment]
Looks good.

**Pull request merged**